### ISSUE TITLE
`basePath` arguments for N5KeyValueReaders/Writers

### ISSUE THREAD

**Issue by bogovicj** [saalfeldlab/n5-aws-s3#20]
@hanslovsky pointed out some strange behavior in the current setup [here](https://imagesc.zulipchat.com/#narrow/stream/328251-NGFF/topic/Read.20zarr.20from.20AWS.20S3.20with.20Java.20N5/near/370911574) (thanks Philipp!).  This issue discusses these points:

> When using AWS SDK I can only successfully list the contents of store/ in the zarr-demo bucket if I omit the leading slash (i.e. store/ works but not /store)
> ...
> I should be able to create ZarrKeyValueReader with or without leading slash. Trailing slash should probably be significant because it is significant in AWS S3 (but not on the file system)

Agree that with or without leading slash should work.  There's a consequence to this:

**`basePath` is relative to the bucket root, whether or not it starts with a leading slash** 

This is different behavior than for local filesytems, where a leading slash indicates the path is absolute, otherwise the path is relative.  @cmhulbert  and I talked, and think this is fine. 

To summarize with examples, these three `basePath`s will be identical:
* "store"
* "/store"
* "/store/"
* "store/"

This should work as of https://github.com/saalfeldlab/n5-aws-s3/pull/16/commits/fd10ee5b9db1af61194a6e73c6346e6ba3e0540e. Note that the `basePath`s below are different
* "./store"
* "//store"

see https://github.com/saalfeldlab/n5-aws-s3/issues/19
